lib/trino/client/query.rb

```diff
@@ -102,12 +104,12 @@ def columns
 
     def column_value_parsers
       @column_value_parsers ||= columns.map {|column|
-        ColumnValueParser.new(column)
+        ColumnValueParser.new(column, scalar_parser)
```

Comment:
`scalar_parser` is replaceable but looks like `ColumnValueParser` won't be recreated once instanciated even if `scalar_parser` is replaced.

```ruby
client.query("select * from sys.node") do |q|
  q.each_row {|row|
    q.scalar_parser = scalar_parser1
    p q.transform_row(row)

    # Expect scalar_parser2 is used but scalar_parser1 is used?
    q.scalar_parser = scalar_parser2
    p q.transform_row(row)
  }
end
```
I know this example is artificial but behavior is counterintuitive.

Reply:
@takezoe great point! Agreed that this likely wouldn't happen, but that it could be a problem for someone who needs it. I have updated this in 035679d while keeping the memoization intact. Now the `scalar_parser` can be changed and I updated the tests to show this.

Reply:
Looks very good. Thank you!